pkg/model/status/Status.go

```diff
@@ -0,0 +1,53 @@
+package status
+
+type Status string
+
+var enums []string
+
+func New(str string) Status {
+
+	for _, s := range enums {
+		if s == str {
+			return Status(s)
+		}
+	}
+
+	panic("Unknown Status: " + str)
+}
+
+func (kind Status) String() string {
+	return string(kind)
+}
+
+func List() []string {
+	return enums
+}
+
+func ListGeneric() []interface{} {
+
+	result := make([]interface{}, len(enums))
+
+	for i, s := range enums {
+		result[i] = s
+	}
+
+	return result
+}
+
+func createStatus(s string) Status {
+
+	enums = append(enums, s)
+
+	return Status(s)
+}
+
+var (
+	Unknown = createStatus("")
```

Comment:
Seeing as I don't need the List or ListGeneric functionality in this solution, constants would be a simpler and safer option. I am always looking for better enum patterns in Go, since it doesn't have a built-in type for it. I found this pattern on a mailinglist a while back, but I never considered it 100% correct.